downstairs/src/lib.rs

```diff
@@ -1736,12 +1944,127 @@ impl Downstairs {
                 };
 
                 Ok(Some(Message::FlushAck {
+                    upstairs_id: job.upstairs_connection.upstairs_id,
+                    session_id: job.upstairs_connection.session_id,
+                    job_id,
+                    result,
+                }))
+            }
+            IOop::ExtentClose {
+                dependencies: _,
+                extent,
+            } => {
+                println!("Closing extent {}", extent);
+                let result = if !self.is_active(job.upstairs_connection) {
+                    error!(self.log, "Upstairs inactive error");
+                    Err(CrucibleError::UpstairsInactive)
+                } else if let Some(ext) = self.region.extents.get_mut(*extent) {
+                    ext.close().await
+                } else {
+                    Err(CrucibleError::InvalidExtent)
+                };
+
+                Ok(Some(Message::ExtentLiveCloseAck {
+                    upstairs_id: job.upstairs_connection.upstairs_id,
+                    session_id: job.upstairs_connection.session_id,
+                    job_id: job.ds_id,
+                    result,
+                }))
+            }
+            IOop::ExtentFlushClose {
+                dependencies: _,
+                extent,
+                flush_number,
+                gen_number,
+                source_downstairs: _,
+                repair_downstairs: _,
+            } => {
+                println!("Flush then Closing extent {}", extent);
```

Comment:
slog instead?

Reply:
fixed in 02d694383914e8a615080888c1b1bcf9de1be8a3